### PR TITLE
Rework Grafana Agent static mode docs to align with flow mode

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

Update Grafana Agent in static mode documentation to align with the general Grafana docs and with flow mode docs

Changes: 

1. Set topic order to match flow mode
1. Add missing Install on Kubernetes topic
1. Remove extra info from general install topic `_index.md`
1. Pull in some of the suggestions from https://github.com/grafana/agent/pull/2947
1. Clean up xrefs and linking
1. Fix up some mistakes and typos in the Flow docs
1. Update Flow and Static topics to align in layout and content 
1. Migrate content into Install, Startup tipics

TODO (possibly candidates for new PRs):
1. Clean up the initial config info. It's currently scattered between dedicated config topics and the original install topics
1. Add missing start/stop for Docker and Kubernetes
1. Add missing uninstall for Docker and Kubernetes
1. Validate the Kubernetes install topic for Static
1. Rework the config topics for Flow and Static
1. Compare the current reworked layout with the similar ongoing efforts in other doc topic areas (handled by other TWs)

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by clayton-cornell** [comment]
@grafana/grafana-agent-maintainers Can someone take a look through this PR - it's getting big and I'd like to try merging and then iterate again on a new PR.

Specific thing that really needs a check - the Kubernetes install topic in Static. I pulled this from what I read on GitHub, but I'm not sure if it's accurate or if it's missing something. I haven't been able to validate.

The rest is a general cleanup and alignment between Flow and Static and starting to align with the rest of the Grafana doc architecture. The cleanup is also focused on splitting topics into one task per topic - like install/uninstall only and pull start/stop and config into separate topics. In the static docs this is going to take a few attempts to untangle things.

File: docs/sources/flow/setup/install/windows.md
Comment (rfratto) [inline docs/sources/flow/setup/install/windows.md:12]: Is "standard install" well known to mean graphical? If so, this is fine.

File: docs/sources/flow/setup/start-agent.md
Comment (rfratto) [inline docs/sources/flow/setup/start-agent.md:153]: $0.02: I feel like simplifying BINARY_PATH to BINARY might be more ambiguous, especially since it's expected to be the path to the binary.

I'll leave it up to you to decide what to put here though, just wanted to raise my concern.

File: docs/sources/flow/setup/start-agent.md
Comment (rfratto) [inline docs/sources/flow/setup/start-agent.md:154]: Hm, maybe I'm not sure I'm following what "Include the path to the configuration file" means. What does a value of CONFIG_FILE mean which "includes the path?"

File: docs/sources/static/configuration/create-config-file.md
Comment (rfratto) [inline docs/sources/static/configuration/create-config-file.md:45]: nit: Should this say "The WAL is required?"

File: docs/sources/static/set-up/install/_index.md
Comment (rfratto) [inline docs/sources/static/set-up/install/_index.md:32]: Should static mode be linking out to this?

Also: we started referring to the operator as "static mode Kubernetes operator," this might be an outdated usage of the previous term here.

File: docs/sources/static/set-up/install/install-agent-docker.md
Comment (rfratto) [inline docs/sources/static/set-up/install/install-agent-docker.md:51]: For consistency with other guides, should we capitalize the drive name here?

```suggestion
  -v WAL_DATA_DIRECTORY:C:\etc\grafana-agent\data ^
  -v CONFIG_FILE_PATH:C:\etc\grafana-agent ^
```

File: docs/sources/static/set-up/install/install-agent-linux.md
Comment (rfratto) [inline docs/sources/static/set-up/install/install-agent-linux.md:86]: I can't remember, but weren't we supposed to use the form "(Optional)" for optional steps?

**Review by rfratto — APPROVED** [review]
LGTM!

File: docs/sources/flow/setup/start-agent.md
Comment (clayton-cornell) [reply docs/sources/flow/setup/start-agent.md:153]: Is it only the path? or the path plus binary?

File: docs/sources/static/configuration/create-config-file.md
Comment (clayton-cornell) [reply docs/sources/static/configuration/create-config-file.md:45]: Fixed in the next commit.

File: docs/sources/static/set-up/install/install-agent-docker.md
Comment (clayton-cornell) [reply docs/sources/static/set-up/install/install-agent-docker.md:51]: Fixed in the next commit.

File: docs/sources/flow/setup/start-agent.md
Comment (rfratto) [reply docs/sources/flow/setup/start-agent.md:153]: Well, it's the path based on the working directory, so it could be a full path or a relative path.

Comment (rfratto) [reply docs/sources/flow/setup/start-agent.md:153]: It's best practice for it to be an absolute path to where the binary is, I guess. 🤷

File: docs/sources/static/set-up/install/install-agent-linux.md
Comment (clayton-cornell) [reply docs/sources/static/set-up/install/install-agent-linux.md:86]: This was a point that was raised recently within the Style Council. The Google Style guide says `Optional:` https://developers.google.com/style/procedures#optional-steps  Looking again, there wasn't a final decision to use `(Optional)` or `Optional:`  The style guide still says to use brackets. I'll follow up on this.

File: docs/sources/static/set-up/install/_index.md
Comment (clayton-cornell) [reply docs/sources/static/set-up/install/_index.md:32]: This is from the original Static docs. I can remove it... or rename it. I'll remove it fro now and if we need to we can add it back in in another iteration.

File: docs/sources/flow/setup/start-agent.md
Comment (clayton-cornell) [reply docs/sources/flow/setup/start-agent.md:153]: I've reverted this change in the next commit.

File: docs/sources/flow/setup/start-agent.md
Comment (clayton-cornell) [reply docs/sources/flow/setup/start-agent.md:154]: I've reverted this change in the next commit.

File: docs/sources/flow/setup/install/windows.md
Comment (clayton-cornell) [reply docs/sources/flow/setup/install/windows.md:12]: I've reverted this in the next commit.

**Pull request merged**
